superset-frontend/src/components/Menu/SubMenu.tsx

```diff
@@ -195,69 +192,57 @@ const SubMenuComponent: React.FunctionComponent<SubMenuProps> = props => {
-                  return (
-                    <Menu.Item key={tab.label}>
-                      <li
-                        role="tab"
-                        data-test={tab['data-test']}
-                        className={
-                          tab.name === props.activeChild ? 'active' : ''
-                        }
-                      >
-                        <div>
-                          <Link to={tab.url || ''}>{tab.label}</Link>
-                        </div>
-                      </li>
-                    </Menu.Item>
-                  );
-                }
-
+        {props.name && <div className="header">{props.name}</div>}
+        <Menu mode={showMenu} style={{ backgroundColor: 'transparent' }}>
+          {props.tabs &&
+            props.tabs.map(tab => {
+              if ((props.usesRouter || hasHistory) && !!tab.usesRouter) {
                 return (
                   <Menu.Item key={tab.label}>
                     <li
-                      className={cx('no-router', {
-                        active: tab.name === props.activeChild,
-                      })}
                       role="tab"
+                      data-test={tab['data-test']}
+                      className={tab.name === props.activeChild ? 'active' : ''}
                     >
-                      <a href={tab.url} onClick={tab.onClick}>
-                        {tab.label}
-                      </a>
+                      <div>
+                        <Link to={tab.url || ''}>{tab.label}</Link>
+                      </div>
                     </li>
                   </Menu.Item>
                 );
-              })}
-          </Menu>
-        </Col>
-        <Col lg={8} md={navRightCol} sm={24} xs={24}>
-          <div className={navRightStyle}>
-            {props.buttons?.map((btn, i) => (
-              <Button
-                key={i}
-                buttonStyle={btn.buttonStyle}
-                onClick={btn.onClick}
-                data-test={btn['data-test']}
-              >
-                {btn.name}
-              </Button>
-            ))}
-          </div>
-        </Col>
+              }
+
+              return (
+                <Menu.Item key={tab.label}>
+                  <li
+                    className={cx('no-router', {
+                      active: tab.name === props.activeChild,
+                    })}
+                    role="tab"
+                  >
+                    <a href={tab.url} onClick={tab.onClick}>
+                      {tab.label}
+                    </a>
+                  </li>
+                </Menu.Item>
+              );
+            })}
+        </Menu>
+        <div className={navRightStyle}>
+          {props.buttons?.map((btn, i) => (
+            <Button
+              key={i}
+              buttonStyle={btn.buttonStyle}
+              onClick={btn.onClick}
+              data-test={btn['data-test']}
```

Comment:
You can query by button `role` and `name` here. `data-test` is not required.

Reply:
I can look into this as a follow up ticket but this is original logic.